clients/src/test/java/org/apache/kafka/clients/producer/KafkaProducerTest.java

```diff
@@ -2263,7 +2263,10 @@ public void configure(Map<String, ?> configs) {
         private Sender sender = mock(Sender.class);
         private TransactionManager transactionManager = mock(TransactionManager.class);
         private Partitioner partitioner = mock(Partitioner.class);
-        private KafkaThread ioThread = mock(KafkaThread.class);
+        private KafkaThread ioThread = new KafkaThread("Fake Kafka Producer I/O Thread", new Runnable() {
+            @Override
+            public void run() {}
+        }, true);
```

Comment:
Is it possible to set up the mock in such a way, so that it can be used? 
Are there native methods involved? That woulod be a reason to not use a mock since Mockito cannot mock native methods.

Reply:
I have run the test again without this change and it looks like it is no longer needed. I see that mockito has been upgraded to newer releases a few times since this PR was created so I'm guessing that has fixed this problem.

I'll push an amended commit to remove this change but let me know if the kafka developers prefer adding a new commit to remove it instead.

Reply:
Done. Amended original commit.